plugins/org.jkiss.dbeaver.data.gis.view/src/org/jkiss/dbeaver/ui/gis/panel/GISBrowserViewer.java

```diff
@@ -141,6 +148,29 @@ public void primeEditorValue(@Nullable Object value) throws DBException
         leafletViewer.setGeometryData(geometries.toArray(new DBGeometry[0]));
     }
 
+    @Override
+    public void dispose() {
+        // Edge uses callbacks which have a lowest priority in UI
+        // So if dispose is sent during that operation this will lead to initialization
+        // on already disposed composite, we don't want this at all
+        // We can prevent this error by delaying dispose in independent thread operation to allow Edge to finish its
+        // initialization to be disposed properly
+        //FIXME That should be removed as soon as Edge will be fixed, this is an awfull hack
+        if (leafletViewer.getBrowser() != null) {
+            super.dispose();
+        } else {
+            new Job("Disposing browser") {
+                @Override
+                protected IStatus run(IProgressMonitor monitor) {
+                    while (leafletViewer.isBrowserCreating()) {
```

Comment:
`isBrowserCreating` will always be `false` here because it's assigned inside constructor.

Reply:
Actually, in that case, it may not be not false. Please look at the comments before this line.